lib/OptimizationNLPModels/src/OptimizationNLPModels.jl

```diff
@@ -0,0 +1,38 @@
+module OptimizationNLPModels
+
+using Reexport
+@reexport using NLPModels, Optimization, ADTypes
+
+function SciMLBase.OptimizationFunction(
+        nlpmodel::AbstractNLPModel; kwargs...)
+    f(x, p) = NLPModels.obj(nlpmodel, x)
```

Comment:
Mostly looks good, for the derivatives though I think we might want to have a choice whether to use the ones from the NLPModel or generate them in Optimization, so we want to use `instantiate_function` here behind a conditional

Reply:
Ah I see in the tests how you think this way can be made to work, that's much more convenient, unless for some c-wrapping reason it's inefficient..

Reply:
To be honest I couldn't really comment on how this would interact with a C wrapper. I implemented it like because it made sense to use the similar things already implemented on `NLPModel` and use defaults for anything not included there

Reply:
We can figure this out once the basic test here starts working